### DISCUSSION TITLE
Single source shortest path to subset of nodes

### DISCUSSION THREAD

**Discussion by saibalmars** [networkit/networkit#689] [Ideas]
Hi, this is more like a feature request for the current SSSP interface.

I have the demand for computing the shortest paths(SP) on a weighted graph _from a single source to multiple target nodes_ (<< graph node size and all within several hops). Currently, I'm computing them by just run the Dijkstra APSP, but it is slower for big graphs, I already tried the latest supported Bidirectional Dijkstra, but in my case, it's even slower than APSP. 

So, I'm wondering can we have some extra feature for the current SSSP interface as follows:
- Can we have the SP computation interface to support SSSP from one node to a list of target nodes instead of what SSSP has right now for just one target node?
- Or, can we set up a cutoff weight to only report SP to nodes with the distance smaller than the cutoff?

**Comment by avdgrinten** [comment]
Do you really run the *APSP*? Or the Dijkstra SSSP? I would not oppose an interface that supports multiple target nodes (probably in a new class though). You could always post a PR. However, I am not convinced yet that this would be so much faster than the SSSP algorithm.
